File: API/Gymme.Repositories/Entities/WorksheetEntity.cs
using Gymme.Repositories.Base;
using System;
using System.Collections.Generic;
using System.Text;

namespace Gymme.Repositories.Entities
{
    public class WorksheetEntity : EntityBase
    {
        public WorksheetEntity(string description, int practiceId)
        {
            Description = description;
            PracticeId = practiceId;
        }

        public string Description { get; set; }
        public int PracticeId { get; set; }

        internal override IEnumerable<string> CheckIntegrity()
        {
            #region .: Error Strings :.

            const string Description_Unvalid_Error = "A descrição deve ter entre 5 e 10 caracteres.";

            #endregion

            if (Description.Length > 4 || Description.Length < 11)
                yield return Description_Unvalid_Error;
        }
    }
}

File: APP/app/src/main/java/com/gymme/data/data/Base/WorksheetEntity.kt
package com.gymme.data.data.Base

import com.gymme.domain.entities.Exercise
import com.gymme.domain.entities.Worksheet
import java.io.Serializable

data class WorksheetEntity (
    var id: Int,
    val description: String,
    val practiceId: Int
) : Serializable


data class InsertWorksheetExercisesRequest (
        var worksheet: Worksheet,
        var exerciseList: List<Exercise>
) : Serializable